src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -80,22 +85,62 @@ private void EventSource_AnyEventRaised(object sender, BuildEventArgs e)
         }
     }
 
-    private readonly Dictionary<string, TimeSpan> _stats = new Dictionary<string, TimeSpan>();
+    private readonly Dictionary<string, TimeSpan> _statsAnalyzers = new Dictionary<string, TimeSpan>();
 
     private void EventSource_BuildFinished(object sender, BuildFinishedEventArgs e)
     {
-        _stats.Merge(buildCheckManager.CreateTracingStats(), (span1, span2) => span1 + span2);
-        string msg = string.Join(Environment.NewLine, _stats.Select(a => a.Key + ": " + a.Value));
-
 
         BuildEventContext buildEventContext = e.BuildEventContext ?? new BuildEventContext(
             BuildEventContext.InvalidNodeId, BuildEventContext.InvalidTargetId,
             BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTaskId);
 
         LoggingContext loggingContext = loggingContextFactory.CreateLoggingContext(buildEventContext);
 
-        // Tracing: https://github.com/dotnet/msbuild/issues/9629
-        loggingContext.LogCommentFromText(MessageImportance.High, msg);
+        if (_areStatsEnabled)
+        {
+            _statsAnalyzers.Merge(buildCheckManager.CreateAnalyzerTracingStats()!, (span1, span2) => span1 + span2);
+            LogAnalyzerStats(loggingContext);
+        }
+    }
+    
+    private void LogAnalyzerStats(LoggingContext loggingContext)
+    {
+        string infraStatPrefix = "infrastructureStat_";
+
+        Dictionary<string, TimeSpan> infraStats = new Dictionary<string, TimeSpan>();
+        Dictionary<string, TimeSpan> analyzerStats = new Dictionary<string, TimeSpan>();
+
+        foreach (var stat in _statsAnalyzers)
+        {
+            if (stat.Key.StartsWith(infraStatPrefix))
+            {
+                string newKey = stat.Key.Replace(infraStatPrefix, string.Empty);
+                infraStats[newKey] = stat.Value;
+            }
+            else
+            {
+                analyzerStats[stat.Key] = stat.Value;
+            }
+        }
+
+        loggingContext.LogCommentFromText(MessageImportance.High, $"BuildCheck run times{Environment.NewLine}");
```

Comment:
Btw. as discussed here: https://github.com/dotnet/msbuild/pull/9853#discussion_r1529349458, the stats might rather be part of analysis, without an opt-in.

The opt-in migh be used to just flip the importance of the messages (default should be `MessageImportance.Low` - so that it's not part of console/file logs under default verbosity; the current opt-in can promote that to `MessageImportance.High` to make it visible in console even with the default verbosity)